database/migrations/2021_05_02_174300_add_filesize_raw_col.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddFileSizeRawCol extends Migration
+{
+	private const TABLE_NAME = 'photos';
+	private const COLUMN_NAME = 'filesize_raw';
+
+	/**
+	 * Run the migrations.
+	 *
+	 * @return void
+	 */
+	public function up()
+	{
+		Schema::table(self::TABLE_NAME, function (Blueprint $table) {
+			$table->unsignedInteger(self::COLUMN_NAME)->default(0)->after('size');
```

Comment:
It wouldn't be too expensive in terms of lines of code to invoke `filesize` for each photo right here, thus ensuring that the data is actually available. Or would it simply take too much time for people with tens of thousands of photos?

Reply:
Good question, I have no clue how expensive this would be. The main reason why I didn't do it, that I did not know how to do it, i.e. how to receive the current entry from DB, construct the correct path of the photo, etc.

Reply:
I think it might be too expensive, especially for users that have something like 50k pictures, you might get hit my a script time-out...

> I didn't do it, that I did not know how to do it,

The `url` field gives you access to the name of the picture. Then you can just use the `Storage` facade to extract the path: `Storage::path('big/' . $url);`

Reply:
A modified the migration such that it uses the stored size as an rough estimation for the new column